### PR TITLE
Fix: unload cce module when not explicitly called for 

### PR DESCRIPTION
**Pull request by wscullin**
This addresses bug #10597. A work-around to bugs in Cray's mic-knl
module leaves the cce module loaded in the build environment which
may lead to side-effects when building with other compilers.

### CODE REVIEW

File: lib/spack/spack/build_environment.py [hunk 1]
```diff
@@ -716,6 +717,10 @@ def setup_package(pkg, dirty):
                 load_module("cce")
             load_module(mod)
 
+        if os.environ.get("CRAY_CPU_TARGET") == "mic-knl":
+            if "cce" not in pkg.compiler.modules:
+                unload_module("cce")
```
Comment (scheibelp) [inline lib/spack/spack/build_environment.py]: Should the `cce` module not be loaded at all unless it is in `pkg.compiler.modules`?

Comment (wscullin) [reply lib/spack/spack/build_environment.py]: @scheibelp cce is Cray's own compilers and they carry with them a set of binutils. It's loaded above the earlier addition purely to get around a one-line bug in Cray's cray-mic-knl module. Unless users are loading it as a compiler, there's no legitimate reason for it to be in the build environment.

**Comment by s-sajid-ali** [comment]
I'm seeing an error related to this on develop when trying to install diffutils : 
```
configure:4656: $? = 0
configure:4645: /gpfs/mira-home/sajid/packages/spack/lib/spack/env/intel/icc -qversion >&5
icc: command line warning #10006: ignoring unknown option '-qversion'
/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../lib64/crt1.o: In function `_start':
/home/abuild/rpmbuild/BUILD/glibc-2.22/csu/../sysdeps/x86_64/start.S:114: undefined reference to `main'
/opt/cray/pe/cce/8.7.6/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: link errors found, deleting executable `a.out'
/usr/bin/sha1sum: a.out: No such file or directory
configure:4656: $? = 1
configure:4645: /gpfs/mira-home/sajid/packages/spack/lib/spack/env/intel/icc -version >&5
icc: command line warning #10006: ignoring unknown option '-version'
/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../lib64/crt1.o: In function `_start':
/home/abuild/rpmbuild/BUILD/glibc-2.22/csu/../sysdeps/x86_64/start.S:114: undefined reference to `main'
/opt/cray/pe/cce/8.7.6/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: link errors found, deleting executable `a.out'
/usr/bin/sha1sum: a.out: No such file or directory
```
And the configure fails with the following :
```
Please verify that both the operating system and the processor support Intel(R) AVX512F, ADX, AVX512ER, AVX512PF and AVX512CD instructions.
```

When I use the commit in the PR, I see a different error again related to binutils 
```
configure:4656: $? = 0
configure:4645: /gpfs/mira-home/sajid/packages/spack/lib/spack/env/intel/icc -qversion >&5
icc: command line warning #10006: ignoring unknown option '-qversion'
/usr/bin/ld: /usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../lib64/crt1.o: in function `_start':
/home/abuild/rpmbuild/BUILD/glibc-2.22/csu/../sysdeps/x86_64/start.S:114: undefined reference to `main'
/usr/bin/ld: link errors found, deleting executable `a.out'
/usr/bin/sha1sum: a.out: No such file or directory
```
Would the fact that spack-arch tells me I'm on `cray-cnl8-mic_knl` but the compiler wants to use `suse-linux` cause the above error ?

**Comment by adamjstewart** [comment]
Spack's Cray support has been completely overhauled since this PR was opened. Can you check out develop and see if this change is still needed? Pinging @becker33 to take a look.

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you think this PR may still be relevant.

**Pull request closed without merging**